### PR TITLE
Enable Archive storage test for Cassandra

### PR DESCRIPTION
**Pull request by Wise-Wizard**
**Which problem is this PR solving?**

This PR addresses the issue [#5203 ](https://github.com/jaegertracing/jaeger/issues/5203)

**Description of the changes**
This PR addresses the issue [#5203](https://github.com/jaegertracing/jaeger/issues/5203) by integrating the testArchiveTrace function into the common Storage Integration type. Added the Archive Trace Test functionality to remaining Backends including Cassandra and Badger
**How was this change tested?**

The changes were tested by running the following command:

```bash
make test
```

```bash
bash scripts/cassandra-integration-test.sh
```

**Checklist**

- [x] I have read [CONTRIBUTING_GUIDELINES.md](https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md)
- [x] I have signed all commits
- [x] I have added unit tests for the new functionality
- [x] I have run lint and test steps successfully
  - `for jaeger: make lint test`
  - `for jaeger-ui: yarn lint` and `yarn test`

### CODE REVIEW

**Comment by Wise-Wizard** [comment]
Can you please review it once? Initializing with ```f.Options.Get(archiveStorageConfig)```
is showing an Error because in the below code snippet, nsCfg.Enabled is false, tried enabling it in it's declaration but still turned out to be false.
Screenshot 2024-05-05 205703

File: plugin/storage/integration/cassandra_test.go [hunk 1]
```diff
@@ -71,6 +71,10 @@ func (s *CassandraStorageIntegration) initializeCassandra(t *testing.T) {
 	require.NoError(t, err)
 	s.SpanReader, err = f.CreateSpanReader()
 	require.NoError(t, err)
+	s.ArchiveSpanReader, err = f.CreateArchiveSpanReader()
```
Comment (yurishkuro) [inline plugin/storage/integration/cassandra_test.go:74]: you need to pass flags to enable archive storage in L65

Comment (Wise-Wizard) [reply plugin/storage/integration/cassandra_test.go:77]: Done, PTAL if the flags are passed correctly.

**Comment by Wise-Wizard** [comment]
I also had a query regarding Badger, unlike other storages, it does not have the Archive Span Writer/Reader methods nor Options initialization of namespaces other than Primary. Is there some kind of documentation, I can refer to that will help me set up the Archive Trace method for Badger?

File: plugin/storage/integration/cassandra_test.go [hunk 2]
```diff
@@ -38,7 +38,7 @@ func newCassandraStorageIntegration() *CassandraStorageIntegration {
 	s := &CassandraStorageIntegration{
 		StorageIntegration: StorageIntegration{
 			GetDependenciesReturnsSource: true,
-			SkipArchiveTest:              true,
+			SkipArchiveTest:              false,
```
Comment (yurishkuro) [inline plugin/storage/integration/cassandra_test.go:41]: just remove the assignment to re-enable

File: plugin/storage/integration/cassandra_test.go [hunk 3]
```diff
@@ -64,13 +64,20 @@ func (s *CassandraStorageIntegration) initializeCassandraFactory(t *testing.T, f
 func (s *CassandraStorageIntegration) initializeCassandra(t *testing.T) {
 	f := s.initializeCassandraFactory(t, []string{
 		"--cassandra.keyspace=jaeger_v1_dc1",
+		"--cassandra-archive.keyspace=jaeger_v1_dc1",
```
Comment (yurishkuro) [inline plugin/storage/integration/cassandra_test.go:67]: you should use a different name vs. the primary keyspace

File: plugin/storage/cassandra/factory.go [hunk 4]
```diff
@@ -153,7 +153,6 @@ func (f *Factory) Initialize(metricsFactory metrics.Factory, logger *zap.Logger)
 		return err
 	}
 	f.primarySession = primarySession
-
```
Comment (yurishkuro) [inline plugin/storage/cassandra/factory.go:156]: please restore empty line

**Review by yurishkuro — COMMENTED** [review]
lgtm

File: plugin/storage/integration/cassandra_test.go (hunk 3 above)
Comment (Wise-Wizard) [reply plugin/storage/integration/cassandra_test.go:66]: In the config.yaml file, Cassandra main and archive have the same Keyspace configuration (jaeger_v1_dc1), using any other configuration is throwing out an error, refusing to connect?

Comment (yurishkuro) [reply plugin/storage/integration/cassandra_test.go:66]: In the script that prepares Cassandra we may need to run schema creation step twice, with different keyspaces.

Comment (Wise-Wizard) [reply plugin/storage/integration/cassandra_test.go]: Can you PTAL? Is this feasible?

**Review by yurishkuro — APPROVED** [review]
lgtm

**Comment by yurishkuro** [comment]
Thanks!

**Comment by Wise-Wizard** [comment]
> Thanks!

No problem, actually I thought I could work on Badger as well in the same PR was going to do it after Cassandra was done?

**Pull request merged**